homeassistant/components/august/__init__.py

```diff
@@ -403,3 +407,15 @@ def _restore_live_attrs(lock_detail, attrs):
     """Restore the non-cache attributes after a cached update."""
     for attr, value in attrs.items():
         setattr(lock_detail, attr, value)
+
+
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: dr.DeviceEntry
+) -> bool:
+    """Remove august config entry from a device if its no longer present."""
+    data: AugustData = hass.data[DOMAIN][config_entry.entry_id]
+    return any(
```

Comment:
I don't follow this condition. Shouldn't we check that there are no identifiers for other domains or identifiers with a device left in data?

Reply:
Maybe I'm reading `websocket_remove_config_entry_from_device` wrong, but I think it we only need to care about if its ok to remove it for the config_entry passed here.

The check comes down to the identifier[0] matching the domain `identifier[0] == DOMAIN` and then identifier[1] aka device does not exist: `not data.get_device(identifier[1])`

So it is find any identifier in the set where the domain matches where the device no longer exists.  

I'm trying to think of a case where that could fail.  If we somehow have multiple identifiers for the device and one of them is the correct one and one of them is stale data, but august hasn't ever changed the device ids so I think it should be ok

Reply:
That will work 👍   Its safer in case something changes in the future.

Reply:
a255aed1414bb87e3aba476e177f1af6797e54d4